packages/astro/components/ViewTransitions.astro

```diff
@@ -87,8 +88,35 @@ const { fallback = 'animate' } = Astro.props as Props;
 		const doc = parser.parseFromString(html, 'text/html');
 		doc.documentElement.dataset.astroTransition = dir;
 		const swap = () => {
-			document.documentElement.replaceWith(doc.documentElement);
+			// Swap head
+			for(const el of Array.from(document.head.children)) {
+				const id = el.getAttribute(PERSIST_ATTR);
+				const newEl = id && doc.head.querySelector(`[${PERSIST_ATTR}="${id}"]`);
+				// If the element exists in the document already, remove it
+				// from the new document and leave the current node alone
+				if(newEl) {
+					newEl.remove();
+				} else {
+					// Otherwise remove the element in the head. It doesn't exist in the new page.
+					el.remove();
+				}
+			}
```

Comment:
That would diff everything, right? This is not really want we want, we want to only keep nodes marked with `transition:persist`, and links which are implicitly marked as such, by us. I don't see this becoming more complex though.

Reply:
Alright 👍 I thought for `head` we probably want to persist anything as much as default instead of only those marked (similar to the discussion above). But if it's not the case here, we can ignore this for now.